### PR TITLE
Cancellation processing + refactor

### PR DESCRIPTION
**Pull request by JCotton1123**

### CODE REVIEW

File: bin/CancelShell.php [hunk 1]
```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env php
+<?php
+
+require_once("../.config.php");
+require_once("../lib/MindBodyApiClient.php");
+require_once("../vendor/autoload.php");
+
+class CancelShell
```
Comment (JCotton1123) [inline bin/CancelShell.php]: Going to refactor this some and du-duplicate the shared code.

File: bin/CancelShell.php [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        $classDateTimestamp = strtotime($argv[2]);
+        if($classDateTimestamp === false){
+            echo "Invalid relative date supplied.\n";
+            exit(1);
+        }
+        $executionMethod = $argv[3];
+        if($executionMethod === false){
+            echo "Invalid execution method supplied, should be 'report' or 'process'.\n";
+            exit(1);
+        }
```
Comment (JCotton1123) [inline bin/CancelShell.php]: Add validation - verify executionMethod is one of: execute, dry-run

File: bin/CancelShell.php [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                        echo "Skipping client b/c they did not use a monthly unlimited package\n";
+                    continue;
+                }
+
+                $date = new DateTime($visit->Service->ExpirationDate);
+                $date->sub(new DateInterval("P{$daysToRemove}D"));
+
+                echo "Removing {$daysToRemove} days from {$visit->Client->FirstName} {$visit->Client->LastName}, Package {$visit->Service->Name}, Expires {$visit->Service->ExpirationDate}, now {$date->format('Y-m-d\T00:00:00')}\n";
+
+                if ($executionMethod == 'process') {
```
Comment (JCotton1123) [inline bin/CancelShell.php]: Implement the following

**Comment by mjuszczak** [comment]
Other than reverting the removal of promotionsShell, this looks great! Ship it!

**Pull request merged**